gusto/timestepping/split_timestepper.py

```diff
@@ -1,13 +1,161 @@
+            spatial_methods (iter,optional): a list of objects describing the
+                methods to use for discretising transport or diffusion terms
+                for each transported/diffused variable. Defaults to None,
+                in which case the terms follow the original discretisation in
+                the equation.
+            physics_schemes: (list, optional): a list of tuples of the form
+                (:class:`PhysicsParametrisation`, :class:`TimeDiscretisation`),
+                pairing physics parametrisations and timestepping schemes to use
+                for each. Timestepping schemes for physics must be explicit.
+                Defaults to None.
+        """
+
+        if spatial_methods is not None:
+            self.spatial_methods = spatial_methods
+        else:
+            self.spatial_methods = []
+
+        # If we have physics schemes, extract these.
+        if 'physics' in term_splitting:
+            if physics_schemes is None:
+                raise ValueError('Physics schemes need to be specified when applying a physics splitting in the SplitTimestepper')
+            else:
+                # Check that the weights are correct for physics:
+                count = 0
+                if weights is not None:
+                    for idx, term in enumerate(term_splitting):
+                        if term == 'physics':
+                            count += weights[idx]
+                    if count != 1:
+                        raise ValueError('Incorrect weights are specified for the physics schemes in the split timestepper.')
+                self.physics_schemes = physics_schemes
+        else:
+            self.physics_schemes = []
+
+        for parametrisation, phys_scheme in self.physics_schemes:
+            # Check that the supplied schemes for physics are valid
+            if hasattr(parametrisation, "explicit_only") and parametrisation.explicit_only:
+                assert isinstance(phys_scheme, ExplicitTimeDiscretisation), \
+                    ("Only explicit time discretisations can be used with "
+                     + f"physics scheme {parametrisation.label.label}")
+
+        self.term_splitting = term_splitting
+        self.dynamics_schemes = dynamics_schemes
+
+        if weights is not None:
+            self.weights = weights
+        else:
+            self.weights = ones(len(self.term_splitting))
+
+        # Check that each dynamics label in term_splitting has a corresponding
+        # dynamics scheme
+        for term in self.term_splitting:
+            if term != 'physics':
+                assert term in self.dynamics_schemes, f"The {term} terms do not have a specified scheme in the split timestepper"
+
+        # Multilevel schemes are currently not supported for the dynamics terms.
+        for label, scheme in self.dynamics_schemes.items():
+            assert scheme.nlevels == 1, "Multilevel schemes are not currently implemented in the split timestepper"
```

Comment:
And here, maybe by doing:
```suggestion
            assert scheme.nlevels == 1, \
                "Multilevel schemes are not currently implemented in the split timestepper"
```